### DISCUSSION TITLE
[Addon Development Question] Get real URL of user

### DISCUSSION THREAD

**Discussion by imdevarsh** [ScratchAddons/ScratchAddons#3714] [Help] [answered]
I need to get the real URL of the page for an addon.
Let's say that I went to scratch.mit.edu/some-url-that-returns-a-404-error and it returns a 404 Not Found.
I want to know where the user came from to the 404 page, so I want to get `some-url-that-returns-a-404-error` and run some code for that. ~The only workaround I can think of is to make different scripts for each possible case, but I want to know if there is a better way to do this.~ It didn't work.
Thanks, webdev03

**Comment by cobaltt7** [comment, marked as answer]
`location.path`?

Please learn basic js topics before making an addon


________________________________
From: webdev03 ***@***.***>
Sent: Friday, October 29, 2021 9:57:54 PM
To: ScratchAddons/ScratchAddons ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ScratchAddons/ScratchAddons] Get real URL (Discussion #3714)


I need to get the real URL of the page for an addon.
Let's say that I went to scratch.mit.edu/some-url-that-returns-a-404-error and it returns a 404 Not Found.
I want to know where the user came from to the 404 page, so I want to get some-url-that-returns-a-404-error and run some code for that. The only workaround I can think of is to make different scripts for each possible case, but I want to know if there is a better way to do this. It didn't work.
Thanks, webdev03

**Reply by imdevarsh** [reply]
Oh, I didn't know that addons could get location! Thanks

**Reply by cobaltt7** [reply]
You can do anything you can do in the console + use addon apis.

**Comment by WorldLanguages** [comment]
Can you tell us what you're trying to develop? I can't think of any addon that would need to do that.

**Reply by imdevarsh** [reply]
It is finding 404 errors and trying to get the best possible URL that could work, if you want to see the code I have forked this repo webdev03/ScratchAddons and in the smart-404 branch
